code/game/objects/structures/stool_bed_chair_nest/xeno_nest.dm

```diff
@@ -27,6 +32,45 @@
 
 	set_hive_data(src, hivenumber)
 
+	buckling_y = list("[NORTH]" = 27, "[SOUTH]" = -19, "[EAST]" = 3, "[WEST]" = 3)
+	buckling_x = list("[NORTH]" = 0, "[SOUTH]" = 0, "[EAST]" = 18, "[WEST]" = -17)
+
+/obj/structure/bed/nest/afterbuckle(mob/current_mob)
+	. = ..()
+	if(. && current_mob.pulledby)
+		current_mob.pulledby.stop_pulling()
+		resisting = 0 //just in case
+		resisting_ready = 0
+
+	if(buckled_mob == current_mob)
+		current_mob.pixel_y = buckling_y["[dir]"]
+		current_mob.pixel_x = buckling_x["[dir]"]
+		current_mob.dir = turn(dir, 180)
+		current_mob.density = FALSE
+		pixel_y = buckling_y["[dir]"]
+		pixel_x = buckling_x["[dir]"]
+		if(dir == SOUTH)
+			buckled_mob.layer = ABOVE_TURF_LAYER
+			if(ishuman(current_mob))
+				var/mob/living/carbon/human/current_human = current_mob
+				for(var/obj/limb/head/current_mobs_head in current_human.limbs)
+					current_mobs_head.layer = TURF_LAYER
+	else
+		current_mob.pixel_y = initial(buckled_mob.pixel_y)
+		current_mob.pixel_x = initial(buckled_mob.pixel_x)
+		current_mob.density = TRUE
+		if(dir == SOUTH)
+			current_mob.layer = initial(current_mob.layer)
+			if(!ishuman(current_mob))
+				var/mob/living/carbon/human/current_human = current_mob
+				for(var/obj/limb/head/current_mobs_head in current_human.limbs)
+					current_mobs_head.layer =  initial(current_mobs_head.layer)
+		if(!currently_in_destruction)
+			qdel(src)
+		return
```

Comment:
intended to not update icon here?

Reply:
Yes. If buckled_mob != current_mob then this has been called after unbuckle and its slated for removal, no sense updating the icon in this case.